pkg/utils/tests/tests.go

```diff
@@ -33,3 +35,22 @@ func WaitTimeout(t *testing.T) time.Duration {
 	}
 	return DefaultWaitTimeout
 }
+
+// TestInterval is just a sensible poll interval that gives fast tests without
+// risk of spamming
+const TestInterval = 100 * time.Millisecond
+
+// AssertEventually waits for f to return true
+func AssertEventually(t *testing.T, f func() bool) {
+	assert.Eventually(t, f, WaitTimeout(t), TestInterval/2)
+}
+
+// WaitWithTimeout waits for the channel to close (or receive anything) and
+// fatals the test if the default wait timeout is exceeded
+func WaitWithTimeout(t *testing.T, ch <-chan struct{}, failMsg string) {
```

Comment:
Would it make sense to name this one with the `Require` convention to convey that it does a fatal exit?
```suggestion
func RequireSignal(t *testing.T, ch <-chan struct{}, failMsg string) {
```